pages/api/log/index.js

```diff
@@ -0,0 +1,34 @@
+import models from '../../../api/models'
+
+const slackApiUrl = 'https://slack.com/api'
+const botToken = process.env.SLACK_BOT_TOKEN
+const channelId = process.env.SLACK_CHANNEL_ID
+
+const toKV = (obj) => {
+  return obj ? Object.entries(obj).reduce((text, [k, v]) => text + ` ${k}=${v}`, '').trimStart() : '-'
+}
+
+const slackPostMessage = ({ id, level, name, message, env, context }) => {
+  const text = `\`${new Date().toISOString()}\` | \`${id} [${level}] ${name}\` | ${message} | ${toKV(context)} | ${toKV({ os: env.os })}`
+  return fetch(slackApiUrl + '/chat.postMessage', {
+    method: 'post',
+    headers: {
+      Authorization: `Bearer ${botToken}`,
+      'Content-type': 'application/json'
+    },
+    body: JSON.stringify({ channel: channelId, text })
+  })
```

Comment:
Mhh, we may need to buffer requests:

> 
Posting messages | 1 per second | Short bursts >1 allowed. If you attempt bursts, there is no guarantee that messages will be stored or displayed to users. If the burst exceeds available limits, users will see an error message indicating that some messages from your app are not being displayed.
-- | -- | --

-- https://api.slack.com/docs/rate-limits

Reply:
[This article](https://medium.com/slack-developer-blog/handling-rate-limits-with-slacks-apis-f6f8a63bdbdc) from someone working at Slack mentions the Slack SDK which uses a queue. I'll just use that. Better than figuring out how large messages can be since and rolling our own queue imo.